[In progress] This PR adds set_synchronous(), trigger_next_simulation_step(), wait_for_simulation_step_to_end(), set_joint_target_velocities(), get_joint_velocities() in DQ_VrepInterface

juanjqo opened the pull request:
This PR adds in `DQ_VrepInterface` class the following methods:
    - set_synchronous()
    - trigger_next_simulation_step()
    - wait_for_simulation_step_to_end()
    - set_joint_target_velocities()
    - get_joint_velocities()

mmmarinho commented:
@juanjqo Nice!

Just a quick sanity check,  but why’s 
`wait_for_simulation_step_to_end()`
not being tested in the example?

juanjqo commented:
> @juanjqo Nice!
> 
> Just a quick sanity check, but why’s `wait_for_simulation_step_to_end()` not being tested in the example?

Hi @mmmarinho, initially, my intention was not to provide a test example but to provide an example using a kinematic control strategy on a dynamic-enabled scene. Therefore, I considered the method wait_for_simulation_step_to_end() unnecessary at the time. However, it makes sense to test all methods. I updated the example to include it.

Best regards, 

Juancho

PS: (I guess I failed the sanity check :grimacing: :neckbeard:)

mmmarinho commented:
@juanjqo No, you did well. No worries.

If adding that method does not make your example worse, it’s better to add it.

But if it reduces the quality of the example, we might consider something else.

bvadorno commented on interfaces/vrep/DQ_VrepInterface.m:
```diff
@@ -152,6 +172,25 @@ function disconnect_all(obj)
         end
+
+        %% Set Synchronous Mode
+        function set_synchronous(obj,flag)
```
Please, improve the documentation. I wonder if `flag` indicates whether the synchronous mode is enabled or disabled.

bvadorno commented on interfaces/vrep/DQ_VrepInterface.m:
```diff
@@ -488,6 +527,83 @@ function set_joint_target_positions(obj,handles,thetas,opmode)
         end
+
+        %% Get joint velocities
+        function [joint_velocities,retval]=get_joint_velocities(obj,handles,opmode)
```
The documentation must be improved following the DQ Robotic's standards. The function must have a descriptive name, and all parameters and returned values must be duly explained.

The method `get_joint_positions` above has the same issue. Please, use this opportunity to fix it too.

juanjqo replied:
I updated the documentation and the signature of the method. Now, the method matches the C++ signature and has better documentation.  Initially, I implemented the signature 
```matlab
[joint_velocities,retval]=get_joint_velocities(obj,handles,opmode)
```
 following the current style of the class.  Now the signature (and the documentation) is as follows

```matlab
 function joint_velocities = get_joint_velocities(obj,jointnames,opmode)
            % This method gets the joint velocities.
            % Usage:
            %      joint_velocities = get_joint_velocities(jointnames, opmode)   
            %          jointnames: The joint names.
            %          (optional) opmode: The operation mode.
            %
            % Example:
            % jointnames={'LBR4p_joint1','LBR4p_joint2','LBR4p_joint3','LBR4p_joint4',...
            %             'LBR4p_joint5','LBR4p_joint6','LBR4p_joint7'};
            % joint_velocities = get_joint_velocities(jointnames);

               .
               .
               .

              
            end
        end 
```

Please let me know if additional modifications are required.

bvadorno commented on interfaces/vrep/DQ_VrepInterface.m:
```diff
@@ -488,6 +527,83 @@ function set_joint_target_positions(obj,handles,thetas,opmode)
+        %% Get joint velocities
+        function [joint_velocities,retval]=get_joint_velocities(obj,handles,opmode)
+            %% Get joint velocities
+            %%  >> joint_names = {'redundantRob_joint1','redundantRob_joint2','redundantRob_joint3','redundantRob_joint4','redundantRob_joint5','redundantRob_joint6','redundantRob_joint7'};
```
Is this comment necessary?

juanjqo replied:
Not really. I implemented the new methods and their respective documentation following the current style of the class `DQ_VrepInterface`.  That is the reason for those comments. However, I removed them.

bvadorno commented on interfaces/vrep/DQ_VrepInterface.m:
```diff
@@ -488,6 +527,83 @@ function set_joint_target_positions(obj,handles,thetas,opmode)
+            end
+        end     
+
+        %% Set Joint Target Positions
```
Same comment as above.

mmmarinho commented:
@juanjqo @bvadorno 
Writing in the main thread so that this comment isn't lost after those conversions are resolved.

Context by @bvadorno 
> Please update Murilo's email. @mmmarinho, that's probably a good time to use another email. :-)

@juanjqo
I'll change them later myself otherwise the `files changed` in this pull request will get messy, but I'll change all source files at once like so
```
find . -type f -execdir rename 's/murilo@nml.t.u-tokyo.ac.jp/murilomarinho@ieee.org/g' {} +
find . -type f -execdir rename 's/murilo@g.ecc.u-tokyo.ac.jp/murilomarinho@ieee.org/g' {} +
```
if this is run at the source of the repo, all mentions to `murilo@nml.t.u-tokyo.ac.jp` and `murilo@g.ecc.u-tokyo.ac.jp` will be changed to `murilomarinho@ieee.org`.

juanjqo commented:
Hi @bvadorno, I updated the class `DQ_VrepInterface`. However, to comply with the DQ Robotics development guidelines, I implemented the changes you requested only in the new methods this PR is adding. Of course, I'll prepare a different PR to improve/update the documentation of the other methods in the `DQ_VrepInterface` class as soon as possible.

Best regards, 

Juancho

juanjqo commented on interfaces/vrep/DQ_VrepInterface.m:
```diff
@@ -74,7 +83,16 @@
 %
 % Contributors to this file:
-%     Murilo Marques Marinho - murilo@nml.t.u-tokyo.ac.jp
+%     1. Murilo Marques Marinho (murilo@nml.t.u-tokyo.ac.jp)
```
@mmmarinho will modify its email later as he mentioned [here].(https://github.com/dqrobotics/matlab/pull/97#issuecomment-1529202098)

mmmarinho commented:
@juanjqo I'd like to make one comment, would it be possible to resolve the conversations to reduce possible confusion?

bvadorno commented:
> Hi @bvadorno, I updated the class `DQ_VrepInterface`. However, to comply with the DQ Robotics development guidelines, I implemented the changes you requested only in the new methods this PR is adding. Of course, If you agree, I'll prepare a different PR to improve/update the documentation of the other methods in the `DQ_VrepInterface` class as soon as possible. What do you think?
> 
> Best regards,
> 
> Juancho

Hi @juanjqo,

That's indeed a good strategy. Please, go ahead with the other pull request.

All the best,
Bruno
